vendor/blockchain_contracts/src/ethereum/fill_contract_slice.rs

```diff
@@ -0,0 +1,33 @@
+use byteorder::{BigEndian, ByteOrder};
+use web3::types::{Address, U256};
+
+pub trait FillContractSlice {
+    fn fill_contract_slice(&self, buf: &mut [u8]);
+}
+
+impl FillContractSlice for Address {
+    fn fill_contract_slice(&self, buf: &mut [u8]) {
+        buf.copy_from_slice(&self[..]);
+    }
+}
+
+impl FillContractSlice for U256 {
+    fn fill_contract_slice(&self, buf: &mut [u8]) {
+        self.to_big_endian(buf);
+    }
+}
+
+impl FillContractSlice for u32 {
```

Comment:
:wrench: We could give this timestamp the same treatment we gave to SecretHash below.